flink-yarn/src/test/java/org/apache/flink/yarn/WorkerSpecContainerResourceAdapterTest.java

```diff
@@ -106,6 +110,31 @@ public void testMatchVcores() {
 		assertThat(adapter.getWorkerSpecs(containerResource1, strategy), containsInAnyOrder(workerSpec1, workerSpec2));
 		assertThat(adapter.getWorkerSpecs(containerResource2, strategy), contains(workerSpec3));
 		assertThat(adapter.getWorkerSpecs(containerResource3, strategy), contains(workerSpec4));
+
+		final int unitMemMB1 = 150;
+		final int unitVcore1 = 15;
+		final WorkerSpecContainerResourceAdapter adapter1 =
+			new WorkerSpecContainerResourceAdapter(
+				getConfigProcessSpecEqualsWorkerSpec(),
+				minMemMB,
+				minVcore,
+				Integer.MAX_VALUE,
+				Integer.MAX_VALUE,
+				unitMemMB1,
+				unitVcore1,
+				Collections.emptyMap());
+
+		final Resource containerResource4 = Resource.newInstance(150, 15);
+
+		assertThat(adapter1.getWorkerSpecs(containerResource1, strategy), empty());
+		assertThat(adapter1.getWorkerSpecs(containerResource2, strategy), empty());
+
+		assertThat(adapter1.tryComputeContainerResource(workerSpec1).get(), is(containerResource4));
+		assertThat(adapter1.tryComputeContainerResource(workerSpec2).get(), is(containerResource4));
+		assertThat(adapter1.tryComputeContainerResource(workerSpec3).get(), is(containerResource4));
+		assertThat(adapter1.tryComputeContainerResource(workerSpec4).get(), is(containerResource4));
+
+		assertThat(adapter1.getWorkerSpecs(containerResource4, strategy), containsInAnyOrder(workerSpec1, workerSpec2, workerSpec3, workerSpec4));
```

Comment:
I'm not sure whether it is necessary to introduce another adapter. Should be good enough to tune the parameters of existing cases.

I guess the problem is that the original test case did not make its intention explicit.